xonsh/shell.py

```diff
@@ -200,11 +201,16 @@ def __init__(self, execer, ctx=None, shell_type=None, **kwargs):
         self.execer = execer
         self.ctx = {} if ctx is None else ctx
         env = builtins.__xonsh__.env
+
         # build history backend before creating shell
-        builtins.__xonsh__.history = hist = xhm.construct_history(
-            env=env.detype(), ts=[time.time(), None], locked=True
-        )
-        env["XONSH_HISTORY_FILE"] = hist.filename
+        if env.get("XONSH_INTERACTIVE"):
+            builtins.__xonsh__.history = hist = xhm.construct_history(
+                env=env.detype(), ts=[time.time(), None], locked=True
+            )
```

Comment:
could you check with lazyobj?

Reply:
@jnoortheen nice thought but we have no guarantee that history will not be called in the xonsh code in the future. Any developer can add calling the history and we will catching side effects again and without any warnings. 

To add backwards comaptibility I replaced history to DummyHistory in case of not interactive mode.